packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L2StandardBridge.sol

```diff
@@ -0,0 +1,214 @@
+     * @param _l2Token Address of L2 token where withdrawal was initiated.
+     * @param _from Account to pull the deposit from on L2.
+     * @param _to Account to give the withdrawal to on L1.
+     * @param _amount Amount of the token to withdraw.
+     * param _l1Gas Unused, but included for potential forward compatibility considerations.
+     * @param _data Optional data to forward to L1. This data is provided
+     *        solely as a convenience for external contracts. Aside from enforcing a maximum
+     *        length, these contracts provide no guarantees about its content.
+     */
+    function _initiateWithdrawal(
+        address _l2Token,
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint32, // _l1Gas,
+        bytes calldata _data
+    )
+        internal
+    {
+        // When a withdrawal is initiated, we burn the withdrawer's funds to prevent subsequent L2 usage
+        IL2StandardERC20(_l2Token).burn(msg.sender, _amount);
+
+        // Construct calldata for l1TokenBridge.finalizeERC20Withdrawal(_to, _amount)
+        address l1Token = IL2StandardERC20(_l2Token).l1Token();
+        bytes memory message;
+
+        if (_l2Token == Lib_PredeployAddresses.OVM_ETH) {
+            // do we need to require that the l1 token is 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE ? Probably not
+            message = abi.encodeWithSelector(
+                        iOVM_L1StandardBridge.finalizeETHWithdrawal.selector,
+                        _from,
+                        _to,
+                        _amount,
+                        _data
+                    );
+        } else {
+            message = abi.encodeWithSelector(
+                        iOVM_L1ERC20Bridge.finalizeERC20Withdrawal.selector,
+                        l1Token,
+                        _l2Token,
+                        _from,
+                        _to,
+                        _amount,
+                        _data
+                    );
+        }
+
+        // Send message up to L1 bridge
+        sendCrossDomainMessage(
+            l1TokenBridge,
+            0,
+            message
+        );
+
+        emit WithdrawalInitiated(l1Token, _l2Token, msg.sender, _to, _amount, _data);
+    }
+
+    /************************************
+     * Cross-chain Function: Depositing *
+     ************************************/
+
+    /**
+     * @inheritdoc iOVM_L2ERC20Bridge
+     */
+    function finalizeDeposit(
+        address _l1Token,
+        address _l2Token,
+        address _from,
+        address _to,
+        uint256 _amount,
+        bytes calldata _data
+    )
+        external
+        override
+        virtual
+        onlyFromCrossDomainAccount(l1TokenBridge)
+    {
+        // Check the target token is compliant and
+        // verify the deposited token on L1 matches the L2 deposited token representation here
+        if (
+            ERC165Checker.supportsInterface(_l2Token, 0x1d1d8b63) &&
```

Comment:
In the case of a malicious token even a try/catch won't help as that can easily just return out of these functions okay and not be caught. There is a separate discussion about this case in Slack I've cc-ed you on.

Reply:
Can we specify where this magic `0x1d1d8b63` came from/how it's generated?

Reply:
Other than that, I think this approach is sufficient to catch most user-incurred errors, and I think that to @elenadimitrova 's point, catching mistakes is the best we can do here.  There's no way to prevent malicious tokens altogether unless we explicitly require certain bytecode, which we definitely do not want to do.

Reply:
What if the bridge doesn't have minting rights? I feel like instead of ERC165 try/catch would suffice 🤔